File: app/src/main/java/com/stevenfrew/beatprompter/midi/IncomingSongPositionPointerMessage.java
package com.stevenfrew.beatprompter.midi;

public class IncomingSongPositionPointerMessage extends IncomingMessage
{
    IncomingSongPositionPointerMessage(byte[] bytes)
    {
        super(bytes);
    }
    public int getMIDIBeat()
    {
        int firstHalf=mMessageBytes[2];
        firstHalf<<=7;
        int secondHalf=mMessageBytes[1];
        return firstHalf|secondHalf;
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/midi/IncomingSongPositionPointerMessage.kt
package com.stevenfrew.beatprompter.midi

class IncomingSongPositionPointerMessage internal constructor(bytes: ByteArray) : IncomingMessage(bytes) {
    val midiBeat: Int
        get() {
            var firstHalf = mMessageBytes[2].toInt()
            firstHalf = firstHalf shl 7
            val secondHalf = mMessageBytes[1].toInt()
            return firstHalf or secondHalf
        }
}